src/neuroconv/datainterfaces/ecephys/maxwell/maxonedatainterface.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional
+
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ....utils.types import FilePathType, FolderPathType
+
+
+class MaxOneRecordingInterface(BaseRecordingExtractorInterface):
+    """
+    Primary data interface class for converting MaxOne data.
+
+    Using the :py:class:`~spikeinterface.extractors.MaxwellRecordingExtractor`.
+    """
+
+    ExtractorName = "MaxwellRecordingExtractor"
+
+    @staticmethod
+    def auto_install_maxwell_hdf5_compression_plugin(
+        hdf5_plugin_path: Optional[FolderPathType] = None, download_plugin: bool = True
+    ) -> None:
+        """
+        If you do not yet have the Maxwell compression plugin installed, this function will automatically install it.
+
+        Parameters
+        ----------
+        hdf5_plugin_path : string or Path, optional
+            Path to your systems HDF5 plugin library.
+            Uses the home directory by default.
+        download_plugin : boolean, default: True
+            Whether or not to force download of the decompression plugin.
+            It's a very lightweight install but does require an internet connection.
+            This is left as True for seamless passive usage and should not impact performance.
+        """
+        from neo.rawio.maxwellrawio import auto_install_maxwell_hdf5_compression_plugin
+
+        auto_install_maxwell_hdf5_compression_plugin(hdf5_plugin_path=hdf5_plugin_path, force_download=download_plugin)
+
+    def __init__(
+        self,
+        file_path: FilePathType,
+        hdf5_plugin_path: Optional[FolderPathType] = None,
+        download_plugin: bool = True,
+        verbose: bool = True,
+    ) -> None:
+        """
+        Load and prepare data for MaxOne.
+
+        Parameters
+        ----------
+        file_path : string or Path
+            Path to the .raw.h5 file.
+        hdf5_plugin_path : string or Path, optional
+            Path to your systems HDF5 plugin library.
+            Uses the home directory by default.
+        download_plugin : boolean, default: True
+            Whether or not to force download of the decompression plugin.
+            It's a very lightweight install but does require an internet connection.
+            This is left as True for seamless passive usage and should not impact performance.
+        verbose : boolean, default: True
+            Allows verbosity.
+        """
+        if system() != "Linux":
+            raise NotImplementedError(
+                "The MaxOneRecordingInterface has not yet been implemented for systems other than Linux."
+            )
+
+        if not os.environ.get("HDF5_PLUGIN_PATH", ""):
+            hdf5_plugin_path = hdf5_plugin_path or Path.home() / "hdf5_plugin_path_maxwell"
+            if not hdf5_plugin_path.exists():
+                self.auto_install_maxwell_hdf5_compression_plugin(hdf5_plugin_path=hdf5_plugin_path)
+            os.environ["HDF5_PLUGIN_PATH"] = str(hdf5_plugin_path)
```

Comment:
OK, this is one of the oddest formats I've ever worked with...

If this 5 lines are not exactly run like this, and at this level, the CI (even restricted to ubuntu only) runs into sporadic segmentation faults but otherwise does not complain as it usually would about not being able to decompress the values in the HDF5 dataset